src/models/mappers/orderMapper.ts

```diff
@@ -11,3 +11,27 @@ export const toOrder = (dto: any, side: Side = Side.Buy) =>
     orderVolume: 0,
     connectedLimitOrders: []
   });
+
+export const mapToMarketEffectivePrice = (volume: number, orders: Order[]) => {
+  let price: number = 0;
+
+  const volumeSum = orders.reduce((sum, order) => {
+    if (sum < volume) {
```

Comment:
Split by smaller if blocks. You can check similar implementation here: https://github.com/LykkeCity/LykkeTerminalWeb/pull/280/files#diff-6eb40c0bd6acaa940ed31c309095a1c5